common/src/main/java/com/wynntils/screens/chattabs/widgets/ChatTabsWidget.java

```diff
@@ -0,0 +1,132 @@
+
+    public ChatTabsWidget(
+            float x,
+            float y,
+            int width,
+            int height,
+            ChatTab chatTab,
+            float gridDivisions,
+            ChatTabEditingScreen parent) {
+        super((int) x, (int) y, width, height, Component.literal(chatTab.getName()));
+
+        this.chatTab = chatTab;
+        this.gridDivisions = gridDivisions;
+        this.parent = parent;
+
+        this.deleteButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.chatTabsGui.delete")
+                                .withStyle(ChatFormatting.DARK_RED),
+                        (button) -> Services.ChatTab.removeTab(chatTab))
+                .pos((int) (this.getX() + (this.width / this.gridDivisions * 17)), this.getY() + (this.height / 2) - 10)
+                .size((int) (this.width / gridDivisions * 5) - 3, 20)
+                .build();
+        this.moveUpButton = new Button.Builder(
+                        Component.literal("\uD83E\uDC1D"), // Up Button Icon
+                        (button) -> decrementChatTabIndex())
+                .pos((int) (this.getX() + (this.width / this.gridDivisions * 22)), this.getY() + (this.height / 2) - 10)
+                .size((int) (this.width / gridDivisions * 2) - 2, 20)
+                .build();
+        this.moveDownButton = new Button.Builder(
+                        Component.literal("\uD83E\uDC1F"), // Down Button Icon
+                        (button) -> incrementChatTabIndex())
+                .pos((int) (this.getX() + (this.width / this.gridDivisions * 24)), this.getY() + (this.height / 2) - 10)
+                .size((int) (this.width / gridDivisions * 2) - 2, 20)
+                .build();
+    }
+
+    @Override
+    protected void renderWidget(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        PoseStack poseStack = guiGraphics.pose();
+
+        if (this.isMouseOver(mouseX, mouseY)) {
+            RenderUtils.drawRect(
+                    poseStack, CommonColors.GRAY.withAlpha(70), this.getX(), this.getY(), 0, this.width, this.height);
+        }
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        StyledText.fromString(chatTab.getName()),
+                        this.getX() + 4,
+                        this.getY() + (this.height >> 1),
+                        CommonColors.WHITE,
+                        HorizontalAlignment.LEFT,
+                        VerticalAlignment.MIDDLE,
+                        TextShadow.NORMAL);
+
+        deleteButton.render(guiGraphics, mouseX, mouseY, partialTick);
+        moveUpButton.render(guiGraphics, mouseX, mouseY, partialTick);
+        moveDownButton.render(guiGraphics, mouseX, mouseY, partialTick);
+    }
+
+    @Override
+    protected void updateWidgetNarration(NarrationElementOutput narrationElementOutput) {}
+
+    private void incrementChatTabIndex() {
```

Comment:
These 2 methods can probably be merged into 1 `updateChatTabIndex` with a parameter direction being 1 or -1

Reply:
In fact, they can probably just be put into `setChatTabIndex` as something like `int index = Services.ChatTab.getTabIndex(chatTab) + direction;`